Sources/ClientRuntime/Plugins/TelemetryPlugin.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public class TelemetryPlugin: Plugin {
+    private let telemetryProvider: TelemetryProvider
+
+    public init(telemetryProvider: TelemetryProvider) {
+        self.telemetryProvider = telemetryProvider
+    }
+
+    public init(
+        contextManager: TelemetryContextManager? = DefaultTelemetry.defaultContextManager,
+        loggerProvider: LoggerProvider? = DefaultTelemetry.defaultLoggerProvider,
+        meterProvider: MeterProvider? = DefaultTelemetry.defaultMeterProvider,
+        tracerProvider: TracerProvider? = DefaultTelemetry.defaultTracerProvider
+    ) {
+        self.telemetryProvider = BasicTelemetryProvider(
+            contextManager: contextManager!,
+            loggerProvider: loggerProvider!,
+            meterProvider: meterProvider!,
+            tracerProvider: tracerProvider!
+        )
+    }
+
+    public func configureClient(clientConfiguration: ClientConfiguration) {
+        if var config = clientConfiguration as? DefaultClientConfiguration {
+            config.telemetryProvider = self.telemetryProvider
+        }
+    }
+}
+
+private class BasicTelemetryProvider: TelemetryProvider {
+    let contextManager: TelemetryContextManager
+    let loggerProvider: LoggerProvider
+    let meterProvider: MeterProvider
+    let tracerProvider: TracerProvider
+
+    public init(
```

Comment:
Q: Is the access level for this class planned to change to public?

Reply:
No, but it could be exposed in the future. `BasicTelemetryProvider` is only used in the `TelemetryPlugin` currently.

Reply:
I see, if that's the case, let's just make the initializer private to match the class.

Reply:
Changed to strictest access-level modifier`fileprivate`.